authors/apps/articles/views.py

```diff
@@ -97,3 +98,50 @@ def destroy(self, request, slug):
             {"message": "Article is deleted"},
             status=status.HTTP_200_OK
         )
+
+
+class FavouritesView(GenericAPIView):
+    serializer_class = FavouriteSerializer
+    permission_classes = (IsAuthenticated,)
+    renderer_classes = (ArticleJSONRenderer,)
+
+    def get_object(self, **kwargs):
+        article_slug = self.kwargs.get('slug')
+        article_obj = get_object_or_404(Article, slug=article_slug)
+        return article_obj
+
+    def put(self, request, **kwargs):
+        article = self.get_object()
+        user = request.user
+        article_id = article.id
+        fav = Favourites.objects.filter(
+            article_id=article_id, user=user
+        )
+        fav_option = fav.first()
+        fav.delete()
+        counter = article.favouriteCount
+        if (counter == 1):
+            article.favourited = False
+        if (counter >= 1):
+            article.favouriteCount = counter - 1
+            article.save()
+            return Response({
+                "message":
+                    "article has been unfavorited"
+            }, status=200)
+        else:
+            serializer = self.serializer_class(data={}, partial=True)
```

Comment:
@moseskamira Look at the logic and make sure it si consistent. Ensure you don't *repeat yourself*.